### PR TITLE
Fix up our build badges

### LINKED ISSUE #30827: Per job GitHub badges
Azure DevOps now provides per job GitHub badges.

See https://raw.githubusercontent.com/Microsoft/azure-pipelines-agent/master/README.md as an example.

This means the Roslyn readme.md can provide status at the same granularity as the Jenkins badges are reporting.

side note: I don't think the Localization badge for Roslyn has been updated yet.

FYI @jaredpar

### PR DESCRIPTION
**Pull request by jaredpar**
closes #30827

### CODE REVIEW

**Comment by jaredpar** [comment]
@dotnet/roslyn-infrastructure please take a look

File: build/scripts/generate-badges.ps1 [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+param ()
+
+$branchNames = @(
+    'master',
+    'master-vs-deps',
+    'dev16.0-preview2',
+    'dev16.0-preview2-vs-deps')
+
+function Get-AzureBadge($branchName, $jobName, $configName) {
+    $template = "[![Build Status](https://dev.azure.com/dnceng/public/_apis/build/status/dotnet/roslyn/roslyn-CI?label=build&branchname=$branchName&jobname=$jobName&configuration=$configName)]"
```
Comment (agocke) [inline build/scripts/generate-badges.ps1]: It's not really a template if you substitute inline

**Comment by jaredpar** [comment]
Thanks!

File: build/scripts/generate-badges.ps1 (hunk 1 above)
Comment (jaredpar) [reply build/scripts/generate-badges.ps1]: There's nothing that says a template has to be configurable just that it's a pattern https://www.merriam-webster.com/dictionary/template

👅

File: README.md [hunk 2]
```diff
@@ -2,26 +2,37 @@
...
...ws_Desktop_Unit_Tests&configuration=release_32)](https://dev.azure.com/dnceng/public/_build/latest?definitionId=15&branchname=master)|[![Build Status](https://dev.azure.com/dnceng/public/_apis/build/status/dotnet/roslyn/roslyn-CI?label=build&branchname=master&jobname=Windows_Desktop_Unit_Tests&configuration=release_64)](https://dev.azure.com/dnceng/public/_build/latest?definitionId=15&branchname=master)|[![Build Status](https://dev.azure.com/dnceng/public/_apis/build/status/dotnet/roslyn/roslyn-CI?label=build&branchname=master&jobname=Windows_Desktop_Spanish_Unit_Tests&configuration=)](https://dev.azure.com/dnceng/public/_build/latest?definitionId=15&branchname=master)|[![Build Status](https://dev.azure.com/dnceng/public/_apis/build/status/dotnet/roslyn/roslyn-CI?label=build&branchname=master&jobname=Windows_Determinism_Test&configuration=)](https://dev.azure.com/dnceng/public/_build/latest?definitionId=15&branchname=master)|[![Build Status](https://dev.azure.com/dnceng/public/_apis/build/status/dotnet/roslyn/roslyn-CI?label=build&branchname=master&jobname=Windows_Correctness_Test&configuration=)](https://dev.azure.com/dnceng/public/_build/latest?definitionId=15&branchname=master)|
```
Comment (dpoeschl) [inline README.md]: These links all go to the same place, which lists the problems across _all_ builds in that `roslyn-CI` run with no immediate indication of what's what. So when you click on an individual queue here you might think the problems you then see are for that queue, but they may or may not be. If we can't make per-queue links (I don't see how but haven't tried hard), consider adding `&view=logs` to the end of the link URLs so you can see the overview and click the queue you were interested in.

**Pull request merged**